ckan/views/resource.py

```diff
@@ -165,7 +165,6 @@ def download(package_type, id, resource_id, filename=None):
 
     try:
         rsc = get_action(u'resource_show')(context, {u'id': resource_id})
```

Comment:
package_show was at least checking that the id passed is a valid dataset that the user has access to, this would be even better and less expensive:

```suggestion
        rsc = get_action(u'resource_show')(context, {u'id': resource_id})
        if rsc['package_id'] != id:
            raise NotFound
```

Reply:
Great! I thought on a similar idea! I added a more explicit error raise with clear message.